src/app_service/service/keycard/service.nim

```diff
@@ -148,6 +149,45 @@ QtObject:
     if self.doLogging:
       debug "keycardCancelFlow", kcServiceCurrFlow=($self.currentFlow), response=response
 
+  ##########################################################
+  ## Used in test env only, for testing keycard flows
+  proc registerMockedKeycard*(self: Service, cardIndex: int, readerState: int, keycardState: int,
+  mockedKeycard: string, mockedKeycardHelper: string) =
+    if not singletonInstance.localAppSettings.getTestEnvironment():
+      return
```

Comment:
shall we aggressively raise error in that case?

Reply:
Well since nobody should call those methods, raising an error or simply doing nothing (just return) is ok by my side. Current qml cannot call them if a user is in other than the test environment. If anybody builds a new feature calling any of the mocked keycard lib-related methods in other than the test env, won't achieve anything. But I can add an error message ofc.

Reply:
@alaibe errors added :)